tests/mr/device/thrust_allocator_tests.cu

```diff
@@ -36,19 +38,36 @@ struct allocator_test : public mr_ref_test {};
 
 TEST_P(allocator_test, first)
 {
+  rmm::mr::set_current_device_resource(this->mr.get());
```

Comment:
These tests are parameterized over memory resources, but we were never selecting them!

Reply:
Shouldn't this happen in fixture setup?

Reply:
That would seem to make sense, but there are other tests that rely on this same fixture _not_ setting the current device resource.
e.g.

https://github.com/rapidsai/rmm/blob/7ed529fb7b94d5639deb0f04efc08ab7ae7fd045/tests/mr/device/mr_ref_tests.cpp#L71-L88